doc/source/index.rst

```diff
@@ -159,6 +159,41 @@ an instance of :class:`datadog.threadstats.ThreadStats`::
 
 datadog.dogstatsd
 =================
+:mod:`datadog.dogstatsd` is a Python client for DogStatsd that automatically
+buffers submitted metrics and submits them to the server asynchronously or at
+maximum sizes for a packet that would prevent fragmentation.
+
+.. note::
+
+    To ensure that all the metrics are sent to the server, either use the
+    context-managed instance of :class:`~datadog.dogstatsd.base.DogStatsd`
+    or when you are finished with the client perform a manual :code:`flush()`
+    otherwise the buffered data may get de-allocated before being sent.
```

Comment:
```suggestion
    or when you are finished with the client perform a manual :code:`flush()`.
    Otherwise the buffered data may get de-allocated before being sent.
```